src/core/Directus/Filesystem/FilesystemFactory.php

```diff
@@ -42,15 +42,21 @@ public static function createLocalAdapter(Array $config, $rootKey = 'root')
 
     public static function createS3Adapter(Array $config, $rootKey = 'root')
     {
-        $client = S3Client::factory([
+        $options = [
             'credentials' => [
                 'key' => $config['key'],
                 'secret' => $config['secret'],
             ],
             'region' => $config['region'],
             'version' => ($config['version'] ?: 'latest'),
-        ]);
+        ];
 
+        if (isset($config['endpoint'])) {
+          $options['endpoint'] = $config['endpoint'];
+          $options['use_path_style_endpoint'] = true;
```

Comment:
We could set it, but in my opinion, users setting a custom endpoint regularly won't set it to AWS' default domain.
Normally they want to address 'my.aws.acme.com', instead of '*.s3.amazonaws.com'.

This way, users are in complete control over their set endpoint.

Reply:
Thanks for the clarification @mvhirsch!